### ISSUE TITLE
Cannot import 'init'

### ISSUE THREAD

**Issue by DyingEcho** [tartley/colorama#115]
**Specs:**
Python 3.5.2
Running on macOS Sierra Terminal (IDLE doesn't work for me)

Colorama doesn't seem to contain the module 'init'.
This was with the following code:
```
#!/usr/bin/python
# coding=latin-1
...
import sys
import os
...
dir_path = os.path.dirname(os.path.realpath(__file__))
sys.path.insert(0,dir_path + "colorama")
from colorama import init, Fore, Back, Style
```

The error:
```
Traceback (most recent call last):
  File "/path/to/file/code/main.py", line 15, in <module>
    from colorama import Fore, Back, Style
ImportError: cannot import name 'init'
```

**Comment by math2001** [comment]
Have you tried to import other stuff from colorama? Try this:

```python
#!/usr/bin/python
# coding=latin-1

import sys
import os

dir_path = os.path.dirname(os.path.realpath(__file__))
sys.path.insert(0,dir_path + "colorama")
from colorama import Fore, Back, Style
```

Also, check that there's `init` in this list:

```python
import colorama
print(dir(colorama))
```

**Comment by DyingEcho** [comment]
First code got an IOError:
```
Traceback (most recent call last):
  File "/Users/TheoAdmin/Desktop/Code/colortest/Untitled.py", line 9, in <module>
    from colorama import Fore, Back, Style
ImportError: cannot import name 'Fore'
```
`init` was not in the list. Result was:
`['__doc__', '__loader__', '__name__', '__package__', '__path__', '__spec__']`

**Comment by kirbyfan64** [comment]
@DyingEcho Try this:

```python
import colorama
print(colorama.__path__)
```

and make sure Python is importing the right colorama.

**Comment by DyingEcho** [comment]
Got `_NamespacePath(['/Users/TheoAdmin/Desktop/Code/colortest/colorama'])`... which is a folder.
I was not under the impression Python was _able_ to import folders.

**Comment by kirbyfan64** [comment]
@DyingEcho So, if I'm understanding correctly, then the main colorama folder is *inside* that folder?

I think something here isn't right...

**Comment by DyingEcho** [comment]
Perhaps there is an error on my part.
Here’s my filesystem:

```
code
    main.py (1)
    other_files.py
    colorama
        README.RTF
        PKG-INFO
        MANIFEST.in
        setup.cfg
        setup.py (2)
        LICENSE.txt
        CHANGELOG.rst
        demos
            demo.bat
            demo.sh
            demo01.py
            demo02.py
            demo03.py
            demo04.py
            demo05.py
            demo06.py
            demo07.py
            demo08.py
            fixpath.py
        colorama.egg-info
            dependency_links.txt
            PKG_INFO
            SOURCES.txt
            top_level.txt
        colorama
            __init__.py
            ansi.py
            ansitowin32.py
            initialise.py
            win32.py
            winterm.py
```
```
(1) File from which I import colorama
(2) Has been run, did not print anything or ask for input
```

So, there are two folders named `colorama` but NO module named `colorama`.
Is this filesystem correct?

**Comment by wiggin15** [comment]
> I was not under the impression Python was able to import folders.

Python is able to import folders if they have an `__init__.py` file. In this case Python imports a package and not a single module.



> _NamespacePath(['/path/to/file/colortest/colorama'])

I'm not sure what this path is, but it doesn't look like the actual colorama and this is what you are actually importing - which is the root of the problem. Maybe this is an older version, bundled with colortest? Try to see what's inside this folder and remove it if you can.


I am not sure why you have the colorama directory tree along with the code and why you are changing sys.path to use it.  If you install the package with [pip](https://pip.pypa.io/en/stable/) or [easy_install](https://wiki.python.org/moin/EasyInstall) or [setup.py](http://stackoverflow.com/questions/1471994/what-is-setup-py) then you should be able to import the package without having the package next to your code and without changing `sys.path`.

**Comment by kirbyfan64** [comment]
I have a theory.

In your code, you're doing this:

```python
dir_path = os.path.dirname(os.path.realpath(__file__))
sys.path.insert(0,dir_path + "colorama")
```

But this isn't correct! If this file is located in, say, `/path/to/code/main.py`, then `dir_path` will contain `/path/to/code`. When `colorama` is appended to that, then you end up with `/path/to/codecolorama`, which probably isn't what you intended and likely doesn't even exist. (The fix is to use `os.path.join(dir_path, 'colorama')`.)

Since `/path/to/codecolorama` doesn't exist, Python can't import colorama from there. In the end, it looks in the directory housing `main.py`.

Now's the golden part. Since there's a colorama folder in the directory with `main.py`, Python loads up that folder as a [namespace package](https://www.python.org/dev/peps/pep-0420/#examples). That's why `colorama.__path__` says `_NamespacePath`.

**TL;DR:** Don't try to combine paths using `+`. Use `os.path.join`.

**Comment by DyingEcho** [comment]
It works, thanks!
